app/services/postcode_service.rb

```diff
@@ -8,18 +8,24 @@ def lookup(postcode)
     return unless postcode.match(POSTCODE_REGEXP)
 
     postcode_lookup = nil
+
     begin
       # URI encoding only supports ASCII characters
+      # Example response for postcode SE27 0AL:
       ascii_postcode = self.class.clean(postcode)
       Timeout.timeout(5) { postcode_lookup = @pio.lookup(ascii_postcode) }
     rescue Timeout::Error
       Rails.logger.warn("Postcodes.io lookup timed out")
     end
+
     if postcode_lookup && postcode_lookup.info.present?
-      {
+      OpenStruct.new({
         location_code: postcode_lookup.codes["admin_district"],
         location_admin_district: postcode_lookup&.admin_district,
-      }
+        incode: postcode_lookup.incode,
+        outcode: postcode_lookup.outcode,
+        result?: postcode_lookup.outcode.present?,
+      })
```

Comment:
Introduced OpenStruct to begin process of refactoring the response into an Object which can have behaviours such as `.result?`